nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/util/SFTPTransfer.java

```diff
@@ -72,17 +74,65 @@
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.Date;
 import java.util.List;
 import java.util.Locale;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 import static org.apache.nifi.processors.standard.util.FTPTransfer.createComponentProxyConfigSupplier;
 
 public class SFTPTransfer implements FileTransfer {
+    // The following DefaultConfig is used for configuration purposes. There are PropertyDescriptors associated with this class
+    // that allow an operator to choose various algorithms and ciphers to allow to be used for their SSH connections.
+    private static final DefaultConfig DEFAULT_CONFIG = new DefaultConfig();
+
+    private static final Set<String> AVAILABLE_KEY_ALGORITHM_NAMES_SET = Collections.unmodifiableSet(DEFAULT_CONFIG.getKeyAlgorithms().stream()
+            .map(Factory.Named::getName)
+            .collect(Collectors.toSet()));
+
+    private static final Set<String> AVAILABLE_CIPHER_NAMES_SET = Collections.unmodifiableSet(DEFAULT_CONFIG.getCipherFactories().stream()
+            .map(Factory.Named::getName)
+            .collect(Collectors.toSet()));
+
+    private static final Set<String> AVAILABLE_MESSAGE_AUTHENTICATION_CODE_NAMES_SET = Collections.unmodifiableSet(DEFAULT_CONFIG.getMACFactories().stream()
+            .map(Factory.Named::getName)
+            .collect(Collectors.toSet()));
+
+    private static final Set<String> AVAILABLE_KEY_EXCHANGE_ALGORITHM_NAMES_SET = Collections.unmodifiableSet(DEFAULT_CONFIG.getKeyExchangeFactories().stream()
+            .map(Factory.Named::getName)
+            .collect(Collectors.toSet()));
+
+    /**
+     * Converts a set of names into an alphabetically ordered comma separated value list.
+     *
+     * @param factorySetNames The set of names
+     * @return An alphabetically ordered comma separated value list of names
+     */
+    private static String convertFactorySetToString(Set<String> factorySetNames) {
+        return factorySetNames
+                .stream()
+                .sorted()
+                .collect(Collectors.toList())
+                .toString().replaceAll("[\\[\\]]", "");
```

Comment:
Relying on `toString()` for formatting a list does not seem like the safest approach.  The Spring Framework `StringUtils` includes a `collectionToCommaDelimitedString` method that should provide the same functionality.

Reply:
Found an option that doesn't require an external library: Java 8 provides a StringJoiner class that is used by a collector "Collectors.joining".